projects/core/src/lib/components/legend/legend-layer/legend-layer.component.ts

```diff
@@ -1,20 +1,59 @@
-import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
-import { AppLayerModel } from '@tailormap-viewer/api';
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { AppLayerModel, ServiceModel } from '@tailormap-viewer/api';
+import { ServerTypeHelper } from '@tailormap-viewer/map';
 
 @Component({
   selector: 'tm-legend-layer',
   templateUrl: './legend-layer.component.html',
   styleUrls: ['./legend-layer.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LegendLayerComponent {
+export class LegendLayerComponent implements OnChanges {
 
   @Input()
   public layer: AppLayerModel | null = null;
 
+  @Input()
+  public service: ServiceModel | null | undefined = null;
+
   @Input()
   public url: string | null = null;
 
+  public urlWithOptions: string | null = null;
+
+  public srcset = '';
+
   public failedToLoadMessage = $localize `Failed to load legend for`;
 
+  public ngOnChanges() {
+
+    this.urlWithOptions = this.url;
+    this.srcset = '';
+
+    if (this.url) {
+      try {
+        const url = new URL(this.url);
+        if (this.service
+          && (this.service.hiDpiMode === 'geoserver' || (this.service.hiDpiMode === 'auto' && ServerTypeHelper.getFromUrl(this.service.url) === 'geoserver')
```

Comment:
This expression is used multiple times in different places, encapsulate in function `ServerTypeHelper.isHiDpiMode(ServiceModel, ServiceHiDpiMode)`